cmd/feegrant.go

```diff
@@ -0,0 +1,288 @@
+package cmd
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/feegrant"
+	"github.com/spf13/cobra"
+)
+
+// flag for feegrant module
+const (
+	FlagExpiration  = "expiration"
+	FlagPeriod      = "period"
+	FlagPeriodLimit = "period-limit"
+	FlagSpendLimit  = "spend-limit"
+	FlagAllowedMsgs = "allowed-messages"
+)
+
+func feegrantQueryGrantsCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "grants [grantee]",
+		Short: "query things about a grantee",
+		Args:  cobra.MaximumNArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cl := config.GetDefaultClient()
+			pageReq, err := ReadPageRequest(cmd.Flags())
+
+			if err != nil {
+				return err
+			}
+
+			var address types.AccAddress
+			var keyNameOrAddress string
+
+			if len(args) == 0 {
+				keyNameOrAddress = cl.Config.Key
+			} else {
+				keyNameOrAddress = args[0]
+			}
+
+			address, err = cl.AccountFromKeyOrAddress(keyNameOrAddress)
+
+			if err != nil {
+				return err
+			}
+
+			grants, err := cl.QueryFeeGrants(address, pageReq)
+			if err != nil {
+				return err
+			}
+
+			// FIXME: if anyone can tell me why JSON output without this doesn't work gets a 5 OSMO tip
+			// FIXME: seems like there should be no side effects with using the MarshalProto function, but thats not the
+			for grant := range grants {
+				_, err := cl.MarshalProto(grants[grant])
```

Comment:
Try to use the marshal interface proto methods on the codec

Reply:
the interesting thing is that it works with this dummy loop, but it doesn't without it. the loop should theoretically change nothing

Reply:
weird...